RJPalmer/JavaCheckers
# Add Main Menu to JavaCheckers for Improved Navigation and User Experience

This pull request implements a Main Menu for the JavaCheckers project, addressing Feature Request #9. The main menu improves the user experience by providing clear navigation options before starting the game. Key updates include:

- Main Menu Screen: A screen that appears when the application launches, offering options to start a new game, view instructions, or exit.
- Start New Game: Clicking this option initiates the gameplay, transitioning smoothly into the game board.
- View Instructions: Clicking this option displays a popup or separate screen with the rules and instructions for playing Checkers.
- Exit: Clicking this option gracefully closes the application.
- UI/UX Design: Simple buttons and a minimalistic design that align with the overall theme of the game.

### Why This Feature?

- Enhances usability by adding clear options for starting a game, accessing instructions, and exiting.

- Provides a more professional and polished user experience, making the game more user-friendly and intuitive.

### Testing Steps:

1. Clone the repository and switch to the main-menu-feature branch.
2. Compile and run the game using javac Checkers.java and java Checkers.
3. Verify that the main menu appears upon launch.
4. Test each menu option:

- Start New Game: Ensure it starts a new game and transitions to the game board.

- View Instructions: Check that it shows the correct game rules.

- Exit: Confirm that it exits the application correctly.

### Checklist:

- [ ] Main menu displays upon launch.

- [ ] Navigation works for “Start New Game,” “View Instructions,” and “Exit.”

- [ ] The design is minimalistic but user-friendly.

- [ ] (Optional) Add unit tests for menu navigation functionality.

### Linked Issues:

- Resolves #9
